Data/ByteString/Lazy/Internal.hs

```diff
@@ -71,17 +76,36 @@ import GHC.Exts                 (IsList(..))
 
 import qualified Language.Haskell.TH.Syntax as TH
 
+#ifdef HS_BYTESTRING_ASSERTIONS
+import Control.Exception (assert)
+#endif
+
+
 -- | A space-efficient representation of a 'Word8' vector, supporting many
 -- efficient operations.
 --
 -- A lazy 'ByteString' contains 8-bit bytes, or by using the operations
 -- from "Data.ByteString.Lazy.Char8" it can be interpreted as containing
 -- 8-bit characters.
 --
-data ByteString = Empty | Chunk {-# UNPACK #-} !S.ByteString ByteString
+data ByteString = Empty
+#ifndef HS_BYTESTRING_ASSERTIONS
+  | Chunk {-# UNPACK #-} !S.ByteString ByteString
+#else
+  | Chunk_ {-# UNPACK #-} !S.ByteString ByteString
+#endif
```

Comment:
I really like what this achieves, but I think it's a downside that this declaration is now much less accessible. Is there a way to avoid this downside?!

As a last resort, explanatory comments for the `Chunk` and `Chunk_` constructors might help.